rspirv/sr/constants.rs

```diff
@@ -14,102 +14,21 @@
 
 use crate::spirv;
 
-use super::{Token, Type};
+use super::Token;
 
-/// The class to represent a SPIR-V constant.
+/// Represents a SPIR-V constant.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Constant {
-    pub(in crate::sr) c: ConstantEnum,
-}
-
-#[derive(Clone, Debug, PartialEq)]
-pub(in crate::sr) enum ConstantEnum {
+pub enum Constant {
     Bool(bool),
-    I32(i32),
-    U32(u32),
-    F32(f32),
+    // TODO: Change to arbitrary width datatype
+    Integer(u32),
+    Float(f32),
     Composite(Vec<Token<Constant>>),
-    Null(Token<Type>),
+    Null,
     Sampler(spirv::SamplerAddressingMode, u32, spirv::SamplerFilterMode),
     SpecBool(bool),
-    SpecI32(i32),
-    SpecU32(u32),
-    SpecF32(f32),
+    SpecInteger(u32),
+    SpecFloat(f32),
     SpecComposite(Vec<Token<Constant>>),
     SpecOp(spirv::Op, Vec<Token<Constant>>),
 }
-
-impl Constant {
-    pub fn is_bool_constant(&self) -> bool {
```

Comment:
These methods are actually probably useful, I was getting a bit delete-happy at this point...